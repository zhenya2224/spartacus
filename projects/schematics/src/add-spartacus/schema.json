{
  "$schema": "http://json-schema.org/schema",
  "$id": "SpartacusSchematics",
  "title": "Spartacus Schematics",
  "type": "object",
  "properties": {
    "project": {
      "type": "string",
      "description": "The name of the project.",
      "$default": {
        "$source": "projectName"
      }
    },
    "features": {
      "type": "array",
      "uniqueItems": true,
      "items": {
        "enum": [
          "ASM",
          "Import-Export",
          "Saved-Cart",
          "Quick-Order",
          "CDC",
          "CDS",
          "Checkout",
          "Order",
          "Digital-Payments",
          "Administration",
          "Order-Approval",
          "Bulk-Pricing",
          "Image-Zoom",
          "Product-Variants",
          "VC-Configurator",
          "Textfield-Configurator",
          "CPQ-Configurator",
          "Qualtrics",
          "SmartEdit",
          "Store-Finder",
          "Personalization",
          "TMS-GTM",
          "TMS-AEPL",
          "Account",
          "Profile"
        ],
        "type": "string"
      },
      "default": [
        "ASM",
        "Checkout",
        "Order",
        "Import-Export",
        "Saved-Cart",
        "Quick-Order",
        "Image-Zoom",
        "Product-Variants",
        "SmartEdit",
        "Store-Finder",
        "Personalization",
        "Account",
        "Profile"
      ],
      "x-prompt": {
        "message": "Which Spartacus features would you like to set up?\nPlease note that for most Spartacus features to be properly configured, the Account feature is required.",
        "type": "list",
        "items": [
          {
            "value": "ASM",
            "label": "Assisted Services Module"
          },
          {
            "value": "Import-Export",
            "label": "Import/Export"
          },
          {
            "value": "Saved-Cart",
            "label": "Saved Cart"
          },
          {
            "value": "Quick-Order",
            "label": "Quick Order"
          },
          {
            "value": "CDC",
            "label": "Customer Data Cloud Integration"
          },
          {
            "value": "CDS",
            "label": "Context-Driven Services Integration"
          },
          {
            "value": "Checkout",
            "label": "Checkout"
          },
          {
            "value": "Order",
            "label": "Order"
          },
          {
            "value": "Digital-Payments",
            "label": "Digital Payments Integration"
          },
          {
            "value": "Administration",
            "label": "Organization - Adminstration (b2b feature)"
          },
          {
            "value": "Order-Approval",
            "label": "Organization - Order Approval (b2b feature)"
          },
          {
            "value": "Bulk-Pricing",
            "label": "Product - Bulk Pricing (b2b feature)"
          },
          {
            "value": "Product-Variants",
            "label": "Product - Variants"
          },
          {
            "value": "Image-Zoom",
            "label": "Product - Image Zoom"
          },
          {
            "value": "VC-Configurator",
            "label": "Product Configurator - Variant Configurator"
          },
          {
            "value": "Textfield-Configurator",
            "label": "Product Configurator - Textfield Configurator"
          },
          {
            "value": "CPQ-Configurator",
            "label": "Product Configurator - CPQ Configurator (b2b feature)"
          },
          {
            "value": "Qualtrics",
            "label": "Qualtrics"
          },
          {
            "value": "SmartEdit",
            "label": "SmartEdit"
          },
          {
            "value": "Store-Finder",
            "label": "Store Finder"
          },
          {
            "value": "Personalization",
            "label": "Tracking - Personalization"
          },
          {
            "value": "TMS-GTM",
            "label": "Tracking - Tag Management System - Google Tag Manager"
          },
          {
            "value": "TMS-AEPL",
            "label": "Tracking - Tag Management System - Adobe Experience Platform Launch"
          },
          {
            "value": "Account",
            "label": "User - Account"
          },
          {
            "value": "Profile",
            "label": "User - Profile"
          }
        ]
      }
    },
    "lazy": {
      "type": "boolean",
      "description": "Lazy load the Spartacus features.",
      "default": true
    },
    "baseUrl": {
      "type": "string",
      "description": "The url of the OCC backend",
      "default": "https://localhost:9002"
    },
    "occPrefix": {
      "type": "string",
      "description": "The OCC API prefix. E.g.: /occ/v2/"
    },
    "baseSite": {
      "type": "string",
      "description": "A comma-separated list of base site(s) to use with Spartacus"
    },
    "currency": {
      "type": "string",
      "description": "A comma-separated list of currencies to use in Spartacus",
      "default": "USD"
    },
    "language": {
      "type": "string",
      "description": "A comma-separated list of languages to use in Spartacus",
      "default": "en"
    },
    "urlParameters": {
      "type": "string",
      "description": "A comma-separated list that represents the order of site-context parameters. E.g.: ['baseSite', 'language', 'currency']"
    },
    "useMetaTags": {
      "type": "boolean",
      "description": "Configure baseUrl via meta tag in index.html"
    },
    "featureLevel": {
      "type": "string",
      "description": "Configure feature level for application."
    },
    "overwriteAppComponent": {
      "type": "boolean",
      "description": "Overwrite the contents of app.component.html file with the <cx-storefront> tag.",
      "default": true
    },
    "pwa": {
      "type": "boolean",
      "description": "Add PWA support to the project.",
      "default": false
    },
    "ssr": {
      "type": "boolean",
      "description": "Add Server-side rendering support to the project.",
      "default": false
    },
    "theme": {
      "type": "string",
      "description": "Select a style theme to add. E.g: santorini"
    }
  },
  "required": []
}
