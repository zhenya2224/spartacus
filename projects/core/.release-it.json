{
  "git": {
    "requireCleanWorkingDir": true,
    "requireUpstream": false,
    "tagName": "core-${version}",
    "commitMessage": "Bumping core version to ${version}",
    "tagAnnotation": "Bumping core version to ${version}"
  },
  "npm": {
    "publishPath": "./../../dist/core"
  },
  "hooks": {
    "after:version:bump": "cd ../.. && ng build core --configuration production"
  },
  "github": {
    "release": true,
    "assets": [
      "../../docs.tar.gz",
      "../../docs.zip",
      "../../spartacussampledataaddon.1905.zip",
      "../../spartacussampledataaddon.1905.tar.gz",
      "../../spartacussampledata.2005.zip",
      "../../spartacussampledata.2005.tar.gz",
      "../../spartacussampledata.2011.zip",
      "../../spartacussampledata.2011.tar.gz",
      "../../spartacussampledata.2105.zip",
      "../../spartacussampledata.2105.tar.gz"
    ],
    "releaseName": "@spartacus/core@${version}",
    "releaseNotes": "ts-node ../../scripts/changelog.ts --verbose --lib core --to core-${version}"
  }
}
