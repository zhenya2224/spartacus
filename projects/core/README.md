# Spartacus Core Library

The Spartacus Core Library provides core Spartacus features, which are a required building block for other Spartacus libraries.

To quickly add Spartacus libraries to an Angular application, you can use Spartacus schematics: `ng add @spartacus/schematics`. This will setup and install Spartacus libraries to your Angular project. Please check the [official Spartacus schematics documentation](https://sap.github.io/spartacus-docs/schematics/) for all the prerequisites and instruction on how to use Spartacus schematics.
