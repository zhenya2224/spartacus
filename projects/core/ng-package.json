{
  "$schema": "../../node_modules/ng-packagr/ng-package.schema.json",
  "dest": "../../dist/core",
  "lib": {
    "entryFile": "./public_api.ts",
    "umdModuleIds": {
      "@ngrx/effects": "effects",
      "@ngrx/router-store": "fromNgrxRouter",
      "@ngrx/store": "store",
      "i18next": "i18next",
      "i18next-http-backend": "i18nextHttpBackend",
      "angular-oauth2-oidc": "angularOAuth2Oidc"
    }
  }
}
