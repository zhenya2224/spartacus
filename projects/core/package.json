{
  "name": "@spartacus/core",
  "version": "4.1.0-next.0",
  "description": "Spartacus - the core framework",
  "keywords": [
    "spartacus",
    "storefront",
    "angular",
    "framework"
  ],
  "homepage": "https://github.com/SAP/spartacus",
  "repository": "https://github.com/SAP/spartacus/tree/develop/projects/core",
  "license": "Apache-2.0",
  "dependencies": {
    "tslib": "^2.3.0"
  },
  "peerDependencies": {
    "@angular/common": "^12.0.5",
    "@angular/core": "^12.0.5",
    "@angular/platform-browser": "^12.0.5",
    "@angular/router": "^12.0.5",
    "@ngrx/effects": "^12.1.0",
    "@ngrx/router-store": "^12.1.0",
    "@ngrx/store": "^12.1.0",
    "angular-oauth2-oidc": "^10.0.1",
    "i18next": "^20.2.2",
    "i18next-http-backend": "^1.2.2",
    "rxjs": "^6.6.0"
  },
  "publishConfig": {
    "access": "public"
  }
}
