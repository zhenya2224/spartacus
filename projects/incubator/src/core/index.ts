export * from './incubator-core.module';
