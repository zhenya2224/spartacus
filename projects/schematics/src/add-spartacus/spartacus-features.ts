import { Rule, SchematicsException, Tree } from '@angular-devkit/schematics';
import {
  SPARTACUS_CORE,
  SPARTACUS_FEATURES_MODULE,
  SPARTACUS_FEATURES_NG_MODULE,
  SPARTACUS_STOREFRONTLIB,
} from '../shared/constants';
import { addModuleImport } from '../shared/utils/new-module-utils';
import { createProgram, saveAndFormat } from '../shared/utils/program';
import { getProjectTsConfigPaths } from '../shared/utils/project-tsconfig-paths';

/** Migration which ensures the spartacus features are being correctly set up */
export function setupSpartacusFeaturesModule(project: string): Rule {
  return (tree: Tree): Tree => {
    const { buildPaths } = getProjectTsConfigPaths(tree, project);

    if (!buildPaths.length) {
      throw new SchematicsException(
        `Could not find any tsconfig file. Cannot configure ${SPARTACUS_FEATURES_NG_MODULE}.`
      );
    }

    const basePath = process.cwd();
    for (const tsconfigPath of buildPaths) {
      configureSpartacusModules(tree, tsconfigPath, basePath);
    }
    return tree;
  };
}

function configureSpartacusModules(
  tree: Tree,
  tsconfigPath: string,
  basePath: string
): void {
  const { appSourceFiles } = createProgram(tree, basePath, tsconfigPath);

  for (const sourceFile of appSourceFiles) {
    if (
      sourceFile
        .getFilePath()
        .includes(`${SPARTACUS_FEATURES_MODULE}.module.ts`)
    ) {
      [
        `// Auth Core
        AuthModule.forRoot(),`,
        'LogoutModule,',
        'LoginRouteModule,',
      ].forEach((content) => {
        addModuleImport(sourceFile, {
          import: [
            {
              moduleSpecifier: SPARTACUS_CORE,
              namedImports: ['AuthModule'],
            },
            {
              moduleSpecifier: SPARTACUS_STOREFRONTLIB,
              namedImports: ['LogoutModule', 'LoginRouteModule'],
            },
          ],
          content,
        });
      });

      [
        `// Basic Cms Components
        HamburgerMenuModule,`,
        'SiteContextSelectorModule,',
        'LinkModule,',
        'BannerModule,',
        'CmsParagraphModule,',
        'TabParagraphContainerModule,',
        'BannerCarouselModule,',
        'CategoryNavigationModule,',
        'NavigationModule,',
        'FooterNavigationModule,',
        'BreadcrumbModule,',
      ].forEach((content) => {
        addModuleImport(sourceFile, {
          import: [
            {
              moduleSpecifier: SPARTACUS_STOREFRONTLIB,
              namedImports: [
                'HamburgerMenuModule',
                'SiteContextSelectorModule',
                'LinkModule',
                'BannerModule',
                'CmsParagraphModule',
                'TabParagraphContainerModule',
                'BannerCarouselModule',
                'CategoryNavigationModule',
                'FooterNavigationModule',
                'NavigationModule',
                'BreadcrumbModule',
              ],
            },
          ],
          content,
        });
      });

      [
        `// User Core,
        UserTransitional_4_2_Module,`,
        'UserOccTransitional_4_2_Module,',
        `// User UI,
        AddressBookModule,`,
        'PaymentMethodsModule,',
        'NotificationPreferenceModule,',
        'MyInterestsModule,',
        'StockNotificationModule,',
        'ConsentManagementModule,',
        'MyCouponsModule,',
      ].forEach((content) => {
        addModuleImport(sourceFile, {
          import: [
            {
              moduleSpecifier: SPARTACUS_CORE,
              namedImports: [
                'UserTransitional_4_2_Module',
                'UserOccTransitional_4_2_Module',
              ],
            },
            {
              moduleSpecifier: SPARTACUS_STOREFRONTLIB,
              namedImports: [
                'AddressBookModule',
                'PaymentMethodsModule',
                'NotificationPreferenceModule',
                'MyInterestsModule',
                'StockNotificationModule',
                'ConsentManagementModule',
                'MyCouponsModule',
              ],
            },
          ],
          content,
        });
      });

      [
        `// Anonymous Consents Core,
        AnonymousConsentsModule.forRoot(),`,
        `// Anonymous Consents UI,
        AnonymousConsentsDialogModule,`,
        'AnonymousConsentManagementBannerModule,',
      ].forEach((content) => {
        addModuleImport(sourceFile, {
          import: [
            {
              moduleSpecifier: SPARTACUS_CORE,
              namedImports: ['AnonymousConsentsModule'],
            },
            {
              moduleSpecifier: SPARTACUS_STOREFRONTLIB,
              namedImports: [
                'AnonymousConsentManagementBannerModule',
                'AnonymousConsentsDialogModule',
              ],
            },
          ],
          content,
        });
      });

      [
        `// Product Core,
        ProductModule.forRoot(),`,
        'ProductOccModule,',
        `// Product UI,
        ProductDetailsPageModule,`,
        'ProductListingPageModule,',
        'ProductListModule,',
        'SearchBoxModule,',
        'ProductFacetNavigationModule,',
        'ProductTabsModule,',
        'ProductCarouselModule,',
        'ProductReferencesModule,',
        'ProductImagesModule,',
        'ProductSummaryModule,',
        'ProductIntroModule,',
      ].forEach((content) => {
        addModuleImport(sourceFile, {
          import: [
            {
              moduleSpecifier: SPARTACUS_CORE,
              namedImports: ['ProductModule', 'ProductOccModule'],
            },
            {
              moduleSpecifier: SPARTACUS_STOREFRONTLIB,
              namedImports: [
                'ProductCarouselModule',
                'ProductDetailsPageModule',
                'ProductFacetNavigationModule',
                'ProductImagesModule',
                'ProductIntroModule',
                'ProductListingPageModule',
                'ProductListModule',
                'ProductPageEventModule',
                'ProductReferencesModule',
                'ProductSummaryModule',
                'ProductTabsModule',
                'SearchBoxModule',
              ],
            },
          ],
          content,
        });
      });

      [
        `// Cart Core,
        CartModule.forRoot(),`,
        'CartOccModule,',
        `// Cart UI,
        CartComponentModule,`,
        'WishListModule,',
      ].forEach((content) => {
        addModuleImport(sourceFile, {
          import: [
            {
              moduleSpecifier: SPARTACUS_CORE,
              namedImports: ['CartModule', 'CartOccModule'],
            },
            {
              moduleSpecifier: SPARTACUS_STOREFRONTLIB,
              namedImports: ['CartComponentModule', 'WishListModule'],
            },
          ],
          content,
        });
      });

      ['CostCenterOccModule,'].forEach((content) => {
        addModuleImport(sourceFile, {
          import: [
            {
              moduleSpecifier: SPARTACUS_CORE,
              namedImports: ['CostCenterOccModule'],
            },
          ],
          content,
        });
      });

      [
        `// Order,
        OrderOccModule`,
      ].forEach((content) => {
        addModuleImport(sourceFile, {
          import: [
            {
              moduleSpecifier: SPARTACUS_CORE,
              namedImports: ['OrderOccModule'],
            },
          ],
          content,
        });
      });

      [
        `// Page Events,
        NavigationEventModule,`,
        'HomePageEventModule,',
        'CartPageEventModule,',
        'ProductPageEventModule,',
      ].forEach((content) => {
        addModuleImport(sourceFile, {
          import: [
            {
              moduleSpecifier: SPARTACUS_STOREFRONTLIB,
              namedImports: [
                'NavigationEventModule',
                'HomePageEventModule',
                'CartPageEventModule',
                'ProductPageEventModule',
              ],
            },
          ],
          content,
        });
      });

      [
        `// External routes,
      ExternalRoutesModule.forRoot(),`,
      ].forEach((content) => {
        addModuleImport(sourceFile, {
          import: [
            {
              moduleSpecifier: SPARTACUS_CORE,
              namedImports: ['ExternalRoutesModule'],
            },
          ],
          content,
        });
      });

      saveAndFormat(sourceFile);

      break;
    }
  }
}
