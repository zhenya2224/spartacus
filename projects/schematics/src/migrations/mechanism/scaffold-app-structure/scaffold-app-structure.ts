import { noop, Rule, SchematicContext, Tree } from '@angular-devkit/schematics';
import { checkAppStructure } from '../../../shared/utils/lib-utils';
import {
  getDefaultProjectNameFromWorkspace,
  scaffoldStructure,
} from '../../../shared/utils/workspace-utils';

export function scaffoldAppStructure(): Rule {
  return (tree: Tree, context: SchematicContext): Rule => {
    const project = getDefaultProjectNameFromWorkspace(tree);

    const spartacusFeatureModuleExists = checkAppStructure(tree, project);
    if (!spartacusFeatureModuleExists) {
      context.logger.info('Scaffolding the new app structure...');
      context.logger.warn(
        'Please migrate manually the rest of your feature modules to the new app structure: https://sap.github.io/spartacus-docs/reference-app-structure/'
      );
    }

    return spartacusFeatureModuleExists
      ? noop()
      : scaffoldStructure({ project });
  };
}
