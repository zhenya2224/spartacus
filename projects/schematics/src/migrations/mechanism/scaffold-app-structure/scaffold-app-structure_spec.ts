/// <reference types="jest" />

import {
  SchematicTestRunner,
  UnitTestTree,
} from '@angular-devkit/schematics/testing';
import {
  Schema as ApplicationOptions,
  Style,
} from '@schematics/angular/application/schema';
import { Schema as WorkspaceOptions } from '@schematics/angular/workspace/schema';
import { scaffoldAppStructure } from './scaffold-app-structure';

const spartacusModulePath = 'src/app/spartacus/spartacus.module.ts';
const exampleSpartacusModule = `
import { NgModule } from '@angular/core';
import { BaseStorefrontModule } from "@spartacus/storefront";
import {MyFeature} from './my-feature.module';
import { SpartacusConfigurationModule } from './spartacus-configuration.module';
import { SpartacusFeaturesModule } from './spartacus-features.module';

@NgModule({
  declarations: [],
  imports: [
    MyFeature,
    SpartacusFeaturesModule,
    SpartacusConfigurationModule,
    BaseStorefrontModule
  ],
  exports: [BaseStorefrontModule]
})
export class SpartacusModule { }
`;

const featuresModulePath = 'src/app/spartacus/spartacus-features.module.ts';
const exampleFeaturesModule = `
import { NgModule } from '@angular/core';
import { AsmFeatureModule } from './features/asm/asm-feature.module';

@NgModule({
  declarations: [],
  imports: [AsmFeatureModule],
})
export class SpartacusFeaturesModule {}
`;

const configurationModulePath =
  'src/app/spartacus/spartacus-configuration.module.ts';
const exampleConfigurationModule = `import { NgModule } from '@angular/core';
import { provideConfig } from '@spartacus/core';
import { layoutConfig } from '@spartacus/storefront';

@NgModule({
  providers: [provideConfig(layoutConfig)],
})
export class SpartacusConfigurationModule {}
`;

describe('scaffold app structure', () => {
  let appTree: UnitTestTree;
  let schematicRunner: SchematicTestRunner;

  const workspaceOptions: WorkspaceOptions = {
    name: 'workspace',
    version: '0.5.0',
  };

  const appOptions: ApplicationOptions = {
    name: 'schematics-test',
    inlineStyle: false,
    inlineTemplate: false,
    routing: false,
    style: Style.Scss,
    skipTests: false,
    projectRoot: '',
  };

  beforeEach(async () => {
    schematicRunner = new SchematicTestRunner(
      'test',
      require.resolve('../../test/migrations-test.json')
    );

    appTree = await schematicRunner
      .runExternalSchematicAsync(
        '@schematics/angular',
        'workspace',
        workspaceOptions
      )
      .toPromise();
    appTree = await schematicRunner
      .runExternalSchematicAsync(
        '@schematics/angular',
        'application',
        appOptions,
        appTree
      )
      .toPromise();
  });

  describe('When the new app structure does NOT exist', () => {
    it('should create it', async () => {
      const resultTree = await schematicRunner
        .callRule(scaffoldAppStructure(), appTree)
        .toPromise();

      expect(resultTree.read(featuresModulePath)?.toString()).toMatchSnapshot();
      expect(
        resultTree.read(configurationModulePath)?.toString()
      ).toMatchSnapshot();
      expect(
        resultTree.read(spartacusModulePath)?.toString()
      ).toMatchSnapshot();
    });
  });

  describe('When the new app structure is already in place', () => {
    beforeEach(async () => {
      appTree.create(spartacusModulePath, exampleSpartacusModule);
      appTree.create(featuresModulePath, exampleFeaturesModule);
      appTree.create(configurationModulePath, exampleConfigurationModule);
    });

    it('should not touch it', async () => {
      const resultTree = await schematicRunner
        .callRule(scaffoldAppStructure(), appTree)
        .toPromise();

      expect(resultTree.read(featuresModulePath)?.toString()).toMatchSnapshot();
      expect(
        resultTree.read(configurationModulePath)?.toString()
      ).toMatchSnapshot();
      expect(
        resultTree.read(spartacusModulePath)?.toString()
      ).toMatchSnapshot();
    });
  });
});
