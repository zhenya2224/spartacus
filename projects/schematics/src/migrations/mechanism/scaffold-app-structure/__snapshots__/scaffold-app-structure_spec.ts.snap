// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`scaffold app structure When the new app structure does NOT exist should create it 1`] = `
"import { NgModule } from '@angular/core';



@NgModule({
  declarations: [],
  imports: [
  ]
})
export class SpartacusFeaturesModule { }
"
`;

exports[`scaffold app structure When the new app structure does NOT exist should create it 2`] = `
"import { NgModule } from '@angular/core';



@NgModule({
  declarations: [],
  imports: [
  ]
})
export class SpartacusConfigurationModule { }
"
`;

exports[`scaffold app structure When the new app structure does NOT exist should create it 3`] = `
"import { NgModule } from '@angular/core';
import { SpartacusFeaturesModule } from './spartacus-features.module';
import { SpartacusConfigurationModule } from './spartacus-configuration.module';



@NgModule({
  declarations: [],
  imports: [
  
    SpartacusFeaturesModule,
       SpartacusConfigurationModule
  ]
})
export class SpartacusModule { }
"
`;

exports[`scaffold app structure When the new app structure is already in place should not touch it 1`] = `
"
import { NgModule } from '@angular/core';
import { AsmFeatureModule } from './features/asm/asm-feature.module';

@NgModule({
  declarations: [],
  imports: [AsmFeatureModule],
})
export class SpartacusFeaturesModule {}
"
`;

exports[`scaffold app structure When the new app structure is already in place should not touch it 2`] = `
"import { NgModule } from '@angular/core';
import { provideConfig } from '@spartacus/core';
import { layoutConfig } from '@spartacus/storefront';

@NgModule({
  providers: [provideConfig(layoutConfig)],
})
export class SpartacusConfigurationModule {}
"
`;

exports[`scaffold app structure When the new app structure is already in place should not touch it 3`] = `
"
import { NgModule } from '@angular/core';
import { BaseStorefrontModule } from \\"@spartacus/storefront\\";
import {MyFeature} from './my-feature.module';
import { SpartacusConfigurationModule } from './spartacus-configuration.module';
import { SpartacusFeaturesModule } from './spartacus-features.module';

@NgModule({
  declarations: [],
  imports: [
    MyFeature,
    SpartacusFeaturesModule,
    SpartacusConfigurationModule,
    BaseStorefrontModule
  ],
  exports: [BaseStorefrontModule]
})
export class SpartacusModule { }
"
`;
